README.md

```diff
@@ -0,0 +1,1743 @@
+
+<h4 id="config_map">config_map</h4>
+
+
+The effective config of the Agent. SHOULD be omitted if unchanged since last reported.
+
+MUST be set if the Agent has received the ReportEffectiveConfig flag in the DataForAgent message.
+
+See AgentConfigMap message definition in the [Configuration](#configuration) section.
+
+<h3 id="remoteconfigstatus-message">RemoteConfigStatus Message</h3>
+
+
+The RemoteConfigStatus message has the following structure:
+
+
+```protobuf
+message RemoteConfigStatus {
+    bytes last_remote_config_hash = 1;
+    enum Status {
+        // Remote config was successfully applied by the Agent.
+        APPLIED = 0;
+
+        // Agent is currently applying the remote config that it received earlier.
+        APPLYING = 1;
+
+        // Agent tried to apply the config received earlier, but it failed.
+        // See error_message for more details.
+        FAILED = 2;
+    }
+    Status status = 2;
+    string error_message = 3;
+}
+```
+
+
+<h4 id="last_remote_config_hash">last_remote_config_hash</h4>
+
+
+The hash of the remote config that was last received by this agent from the management server. The server SHOULD compare this hash with the config hash it has for the agent and if the hashes are different the server MUST include the remote_config field in the response in the DataForAgent message.
+
+<h4 id="status">status</h4>
+
+
+The status of the Agent's attempt to apply a previously received remote configuration.
+
+<h4 id="error_message">error_message</h4>
+
+
+Optional error message if status==FAILED.
+
+<h3 id="agentaddonstatuses-message">AgentAddonStatuses Message</h3>
+
+
+The AgentAddonStatuses message describes the status of all addons that the agent has or was offered. The message has the following structure:
+
+
+```protobuf
+message AgentAddonStatuses {
+    map<string, AgentAddonStatus> addons = 1;
+    bytes server_provided_all_addons_hash = 2;
+}
+```
+
+
+<h4 id="addons">addons</h4>
+
+
+A map of AgentAddonStatus messages, where the keys are addon names. The key MUST match the name field of [AgentAddonStatus](#agentaddonstatus-message) message.
+
+<h4 id="server_provided_all_addons_hash">server_provided_all_addons_hash</h4>
+
+
+The aggregate hash of all addons that this Agent previously received from the server via AddonsAvailable message.
+
+The server SHOULD compare this hash to the aggregate hash of all addons that it has for this Agent and if the hashes are different the server SHOULD send an AddonsAvailable message to the agent.
+
+<h3 id="agentaddonstatus-message">AgentAddonStatus Message</h3>
+
+
+The AgentAddonStatus has the following structure:
+
+
+```protobuf
+message AgentAddonStatus {
+    string name = 1;
+    bytes agent_has_hash = 2;
```

Comment:
Perhaps we could add `string server_offered_version` here as well?

Reply:
Created an issue to decide https://github.com/open-telemetry/opamp-spec/issues/8